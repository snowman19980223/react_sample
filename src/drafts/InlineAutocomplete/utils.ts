import getCaretCoordinates from '@koddsson/textarea-caret'
import {Children, EventHandler, SyntheticEvent} from 'react'

import {Coordinates, ShowSuggestionsEvent, Suggestion, TextInputCompatibleChild, Trigger} from './types'

const singleWordTriggerTerminators = new Set([' ', '\n'])
const multiWordTriggerTerminators = new Set(['.', '\n'])

const isWhitespace = (char: string) => /\s/.test(char)

/**
 * Calculate whether or not suggestions should be shown based on the given state of the
 * input. If they should be shown, returns the show event.
 */
export const calculateSuggestionsQuery = (
  triggers: Array<Trigger>,
  text: string,
  caretLocation: number
): ShowSuggestionsEvent | null => {
  // Build backwards from the caret location until the most recent trigger character or terminator
  for (
    let i = caretLocation - 1, query = '', potentialTriggers = triggers;
    i >= 0 && potentialTriggers.length > 0;
    i--
  ) {
    const character = text[i]

    if (singleWordTriggerTerminators.has(character)) potentialTriggers = potentialTriggers.filter(t => t.multiWord)
    if (multiWordTriggerTerminators.has(character)) potentialTriggers = potentialTriggers.filter(t => !t.multiWord)

    for (const trigger of potentialTriggers.filter(t => character === t.triggerChar)) {
      // Trigger chars must always be preceded by whitespace or be the first character in the input,
      // and even a multi-word query cannot start with whitespace
      if ((i === 0 || isWhitespace(text[i - 1])) && !isWhitespace(query[0])) return {trigger, query}

      potentialTriggers = potentialTriggers.filter(t => t !== trigger)
    }

    query = character + query
  }

  return null
}

/**
 * Obtain the coordinates (px) of the bottom left of a character in an input, relative to the
 * top-left corner of the input itself.
 * @param input The target input element.
 * @param index The index of the character to calculate for.
 * @param adjustForScroll Control whether the returned value is adjusted based on scroll position.
 */
export const getCharacterCoordinates = (
  input: HTMLTextAreaElement | HTMLInputElement | null,
  index: number,
  adjustForScroll = true
): Coordinates => {
  if (!input) return {top: 0, left: 0}

  // word-wrap:break-word breaks the getCaretCoordinates calculations (a bug), and word-wrap has
  // no effect on input element anyway
  if (input instanceof HTMLInputElement) input.style.wordWrap = ''

  let coords = getCaretCoordinates(input, index)

  // The library calls parseInt on the computed line-height of the element, failing to account for
  // the possibility of it being 'normal' (another bug). In that case, fall back to a rough guess
  // of 1.2 based on MDN: "Desktop browsers use a default value of roughly 1.2".
  if (isNaN(coords.height)) coords.height = parseInt(getComputedStyle(input).fontSize) * 1.2

  // Sometimes top is negative, incorrectly, because of the wierd line-height calculations around
  // border-box sized single-line inputs.
  coords.top = Math.abs(coords.top)

  // For some single-line inputs, the rightmost character can be accidentally wrapped even with the
  // wordWrap fix above. If this happens, go back to the last usable index
  let adjustedIndex = index
  while (input instanceof HTMLInputElement && coords.top > coords.height) {
    coords = getCaretCoordinates(input, --adjustedIndex)
  }

  const scrollTopOffset = adjustForScroll ? -input.scrollTop : 0
  const scrollLeftOffset = adjustForScroll ? -input.scrollLeft : 0

  return {top: coords.top + coords.height + scrollTopOffset, left: coords.left + scrollLeftOffset}
}

/**
 * Obtain the coordinates of the bottom left of a character in an input relative to the top-left
 * of the page.
 * @param input The target input element.
 * @param index The index of the character to calculate for.
 */
export const getAbsoluteCharacterCoordinates = (
  input: HTMLTextAreaElement | HTMLInputElement | null,
  index: number
): Coordinates => {
  const {top: relativeTop, left: relativeLeft} = getCharacterCoordinates(input, index, true)
  const {top: viewportOffsetTop, left: viewportOffsetLeft} = input?.getBoundingClientRect() ?? {top: 0, left: 0}

  return {
    top: viewportOffsetTop + relativeTop,
    left: viewportOffsetLeft + relativeLeft
  }
}

export const getSuggestionValue = (suggestion: Suggestion): string =>
  typeof suggestion === 'string' ? suggestion : suggestion.value

export const getSuggestionKey = (suggestion: Suggestion): string =>
  typeof suggestion === 'string' ? suggestion : suggestion.key ?? suggestion.value

/**
 * Replace a section of a string.
 */
export const replaceSlice = (
  original: string,
  [startInclusive, endExclusive]: [number, number],
  replacement: string
) => {
  const before = original.substring(0, startInclusive)
  const after = original.substring(endExclusive)
  return before + replacement + after
}

/**
 * Attempts to assert that the child element is of a supported type. This can't be enforced
 * by the type system so it has to be done as a runtime check. This isn't foolproof - a
 * component that forwards a ref to a correct element but does not forward event handlers
 * will not work. But it's the best we can reasonably do.
 */
export function requireChildrenToBeInput(
  child: React.ReactElement,
  childRef: React.RefObject<HTMLElement>
): TextInputCompatibleChild {
  Children.only(child) // Assert that the child is lonely
  if (
    // There is no way to know what type the underlying child is until it mounts, so this
    // will always pass on first render before failing on the second render
    childRef.current &&
    !(childRef.current instanceof HTMLInputElement) &&
    !(childRef.current instanceof HTMLTextAreaElement)
  ) {
    throw new TypeError(
      `AutocompleteTextarea child must be a component that forwards a ref and props to an <input> or <textarea> element.`
    )
  }
  return child
}

/**
 * Combine several event handlers into one. The last handler in the list is called first
 * and no further handlers will be called if `event.preventDefault()` is called.
 */
export const augmentHandler =
  <E extends SyntheticEvent>(...handlers: Array<EventHandler<E> | undefined>) =>
  (event: E) => {
    for (const handler of [...handlers].reverse()) {
      if (!event.isDefaultPrevented()) handler?.(event)
    }
  }
