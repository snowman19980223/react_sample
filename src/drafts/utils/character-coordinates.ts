import getCaretCoordinates from '@koddsson/textarea-caret'

export type Coordinates = {
  top: number
  left: number
}

/**
 * Obtain the coordinates (px) of the bottom left of a character in an input, relative to the
 * top-left corner of the input itself.
 * @param input The target input element.
 * @param index The index of the character to calculate for.
 * @param adjustForScroll Control whether the returned value is adjusted based on scroll position.
 */
export const getCharacterCoordinates = (
  input: HTMLTextAreaElement | HTMLInputElement | null,
  index: number,
  adjustForScroll = true
): Coordinates => {
  if (!input) return {top: 0, left: 0}

  // word-wrap:break-word breaks the getCaretCoordinates calculations (a bug), and word-wrap has
  // no effect on input element anyway
  if (input instanceof HTMLInputElement) input.style.wordWrap = ''

  let coords = getCaretCoordinates(input, index)

  // The library calls parseInt on the computed line-height of the element, failing to account for
  // the possibility of it being 'normal' (another bug). In that case, fall back to a rough guess
  // of 1.2 based on MDN: "Desktop browsers use a default value of roughly 1.2".
  if (isNaN(coords.height)) coords.height = parseInt(getComputedStyle(input).fontSize) * 1.2

  // Sometimes top is negative, incorrectly, because of the wierd line-height calculations around
  // border-box sized single-line inputs.
  coords.top = Math.abs(coords.top)

  // For some single-line inputs, the rightmost character can be accidentally wrapped even with the
  // wordWrap fix above. If this happens, go back to the last usable index
  let adjustedIndex = index
  while (input instanceof HTMLInputElement && coords.top > coords.height) {
    coords = getCaretCoordinates(input, --adjustedIndex)
  }

  const scrollTopOffset = adjustForScroll ? -input.scrollTop : 0
  const scrollLeftOffset = adjustForScroll ? -input.scrollLeft : 0

  return {top: coords.top + coords.height + scrollTopOffset, left: coords.left + scrollLeftOffset}
}

/**
 * Obtain the coordinates of the bottom left of a character in an input relative to the top-left
 * of the page.
 * @param input The target input element.
 * @param index The index of the character to calculate for.
 */
export const getAbsoluteCharacterCoordinates = (
  input: HTMLTextAreaElement | HTMLInputElement | null,
  index: number
): Coordinates => {
  const {top: relativeTop, left: relativeLeft} = getCharacterCoordinates(input, index, true)
  const {top: viewportOffsetTop, left: viewportOffsetLeft} = input?.getBoundingClientRect() ?? {top: 0, left: 0}

  return {
    top: viewportOffsetTop + relativeTop,
    left: viewportOffsetLeft + relativeLeft
  }
}
