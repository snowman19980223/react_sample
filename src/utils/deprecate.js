/* eslint-disable no-console */

export default function({name, message, version}){
  if (process.env.NODE_ENV !== "production") {
    console.warn(`WARNING! ${name} is deprecated and will be removed in version ${version}. ${message}`)
  }
}
