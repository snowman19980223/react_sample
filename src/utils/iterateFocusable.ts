/**
 * Returns an iterator over all of the focusable elements within `container`.
 * Note: If `container` is itself focusable it will be included in the results.
 * @param container The container over which to find focusable elements.
 * @param reverse If true, iterate backwards through focusable elements.
 */
export function* iterateFocusableElements(
  container: Element,
  reverse = false
): Generator<HTMLElement, undefined, never> {
  const walker = document.createTreeWalker(container, NodeFilter.SHOW_ELEMENT, {
    acceptNode: node =>
      node instanceof HTMLElement && node.tabIndex === -1 ? NodeFilter.FILTER_SKIP : NodeFilter.FILTER_ACCEPT
  })
  let nextNode: Node | null = null;

  // If iterating in reverse, continue traversing down into the last child until we reach
  // a leaf DOM node
  if (reverse) {
      let lastChild = walker.lastChild()
      while (lastChild) {
        nextNode = lastChild
        lastChild = walker.lastChild()
      }
  } else {
    nextNode = walker.firstChild()
  }
  while (nextNode instanceof HTMLElement) {
    yield nextNode
    nextNode = reverse ? walker.previousNode() : walker.nextNode()
  }
  return undefined
}
