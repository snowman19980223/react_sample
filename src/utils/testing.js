import renderer from 'react-test-renderer'
import enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

enzyme.configure({adapter: new Adapter()})

/**
 * Render the component (a React.createElement() or JSX expression)
 * into its intermediate object representation with 'type',
 * 'props', and 'children' keys
 *
 * The returned object can be matched with expect().toEqual(), e.g.
 *
 * ```js
 * expect(render(<Foo />)).toEqual(render(<div foo='bar' />))
 * ```
 */
export function render(component) {
  return renderer.create(component).toJSON()
}

/**
 * Get the HTML class names rendered by the component instance
 * as an array.
 *
 * ```js
 * expect(renderClasses(<div className='a b' />))
 *   .toEqual(['a', 'b'])
 * ```
 */
export function renderClasses(component) {
  const {
    props: {className}
  } = render(component)
  return className ? className.trim().split(' ') : []
}

/**
 * console.error() is ugly af in jest; mock it with a noop,
 * and return the mock so that you can restore it:
 *
 * ```js
 * const mock = silenceConsoleError(jest)
 * // do stuff that calls console.error()
 * mock.mockRestore() // restore console.error()
 * ```
 *
 * Note: we need to pass `jest` as a reference here because
 * jest's mocking functions are bound to a module at runtime, and
 * need to be called as methods rather than pure functions
 * exported from the jest-mock module.
 */
export function silenceConsoleError(jest) {
  return jest.spyOn(console, 'error').mockImplementation(jest.fn())
}

export function mount(component) {
  return enzyme.mount(component)
}
