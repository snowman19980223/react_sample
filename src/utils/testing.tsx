import React from 'react'
import {promisify} from 'util'
import renderer from 'react-test-renderer'
import enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import {ThemeProvider} from 'styled-components'
import {default as defaultTheme} from '../theme'

// eslint-disable-next-line @typescript-eslint/no-var-requires
const readFile = promisify(require('fs').readFile)

export const COMPONENT_DISPLAY_NAME_REGEX = /^[A-Z][A-Za-z]+(\.[A-Z][A-Za-z]+)*$/

enzyme.configure({adapter: new Adapter()})

export function mount(component: React.ReactElement) {
  return enzyme.mount(component)
}

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace jest {
    interface Matchers<R> {
      toImplementSystemProps: (systemProps: any) => boolean
      toImplementSxProp: () => boolean
      toImplementSxBehavior: () => boolean
      toSetDefaultTheme: () => boolean
      toSetExports: (exports: Record<string, string>) => boolean
    }
  }
}

/**
 * Render the component (a React.createElement() or JSX expression)
 * into its intermediate object representation with 'type',
 * 'props', and 'children' keys
 *
 * The returned object can be matched with expect().toEqual(), e.g.
 *
 * ```js
 * expect(render(<Foo />)).toEqual(render(<div foo='bar' />))
 * ```
 */
export function render(component: React.ReactElement) {
  return renderer.create(component).toJSON() as renderer.ReactTestRendererJSON
}

/**
 * Render the component (a React.createElement() or JSX expression)
 * using react-test-renderer and return the root node
 * ```
 */
export function renderRoot(component: React.ReactElement) {
  return renderer.create(component).root
}

/**
 * Get the HTML class names rendered by the component instance
 * as an array.
 *
 * ```js
 * expect(renderClasses(<div className='a b' />))
 *   .toEqual(['a', 'b'])
 * ```
 */
export function renderClasses(component: React.ReactElement): string {
  const {
    props: {className}
  } = render(component)
  return className ? className.trim().split(' ') : []
}

/**
 * Returns true if a node renders with a single class.
 */
export function rendersClass(node: React.ReactElement, klass: string): boolean {
  return renderClasses(node).includes(klass)
}

export function renderWithTheme(node: React.ReactElement, theme = defaultTheme): renderer.ReactTestRendererJSON {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>)
}

export function px(value: number | string): string {
  return typeof value === 'number' ? `${value}px` : value
}

export function percent(value: number | string): string {
  return typeof value === 'number' ? `${value}%` : value
}

export function renderStyles(node: React.ReactElement): any {
  const {
    props: {className}
  } = render(node)
  return getComputedStyles(className)
}

export function getComputedStyles(className: string): Record<string, string | Record<string, string>> {
  const div = document.createElement('div')
  div.className = className

  const computed = {} as any
  for (const sheet of document.styleSheets) {
    // CSSRulesLists assumes every rule is a CSSRule, not a CSSStyleRule
    for (const rule of sheet.cssRules) {
      if (rule instanceof CSSMediaRule) {
        readMedia(rule)
      } else if (rule instanceof CSSStyleRule) {
        readRule(rule, computed)
      } else {
        // console.warn('rule.type =', rule.type)
      }
    }
  }

  return computed

  function matchesSafe(node: HTMLDivElement, selector: string) {
    if (!selector) {
      return false
    }
    try {
      return node.matches(selector)
    } catch (error) {
      return false
    }
  }

  function readRule(rule: CSSStyleRule, dest: any) {
    if (matchesSafe(div, rule.selectorText)) {
      const {style} = rule
      for (let i = 0; i < style.length; i++) {
        const prop = style[i]
        dest[prop] = style.getPropertyValue(prop)
      }
    } else {
      // console.warn('no match:', rule.selectorText)
    }
  }

  function readMedia(mediaRule: CSSMediaRule) {
    const key = `@media ${mediaRule.media[0]}`
    // const dest = computed[key] || (computed[key] = {})
    const dest = {}
    for (const rule of mediaRule.cssRules) {
      if (rule instanceof CSSStyleRule) {
        readRule(rule, dest)
      }
    }

    // Don't add media rule to computed styles
    // if no styles were actually applied
    if (Object.keys(dest).length > 0) {
      computed[key] = dest
    }
  }
}
/**
 * This provides a layer of compatibility between the render() function from
 * react-test-renderer and Enzyme's mount()
 */
export function getProps(node: React.ReactElement) {
  return typeof node.props === 'function' ? node.props() : node.props
}

export function getClassName(node: React.ReactElement) {
  return getProps(node).className
}

export function getClasses(node: React.ReactElement) {
  const className = getClassName(node)
  return className ? className.trim().split(/ +/) : []
}

export async function loadCSS(path: string) {
  const css = await readFile(require.resolve(path), 'utf8')
  const style = document.createElement('style')
  style.setAttribute('data-path', path)
  style.textContent = css
  document.head.appendChild(style)
  return style
}

export function unloadCSS(path: string) {
  const style = document.querySelector(`style[data-path="${path}"]`)
  if (style) {
    style.remove()
    return true
  }
}

// If a component requires certain props or other conditions in order
// to render without errors, you can pass a `toRender` function that
// returns an element ready to be rendered.

interface Options {
  skipAs?: boolean
  skipSx?: boolean
}

interface BehavesAsComponent {
  Component: React.ComponentType<any>
  systemPropArray: any[]
  toRender?: () => React.ReactElement
  options?: Options
}

export function behavesAsComponent({Component, systemPropArray, toRender, options}: BehavesAsComponent) {
  options = options || {}

  const getElement = () => (toRender ? toRender() : <Component />)

  if (!options.skipSx) {
    it('implements sx prop behavior', () => {
      expect(getElement()).toImplementSxBehavior()
    })
  }

  if (!options.skipAs) {
    it('respects the as prop', () => {
      const As = React.forwardRef<HTMLDivElement>((_props, ref) => <div className="as-component" ref={ref} />)
      const elem = React.cloneElement(getElement(), {as: As})
      expect(render(elem)).toEqual(render(<As />))
    })
  }

  it('sets a valid displayName', () => {
    expect(Component.displayName).toMatch(COMPONENT_DISPLAY_NAME_REGEX)
  })

  it('sets the default theme', () => {
    expect(getElement()).toSetDefaultTheme()
  })

  it('renders consistently', () => {
    expect(render(getElement())).toMatchSnapshot()
  })
}

export function checkExports(path: string, exports: Record<any, any>): void {
  it('has declared exports', () => {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const mod = require(`../${path}`)
    expect(mod).toSetExports(exports)
  })
}
