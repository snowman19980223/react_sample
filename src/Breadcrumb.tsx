import classnames from 'classnames'
import * as History from 'history'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import Box from './Box'
import {COMMON, FLEX, get, SystemCommonProps, SystemFlexProps} from './constants'
import sx, {SxProp} from './sx'
import theme from './theme'
import {ComponentProps} from './utils/types'

const SELECTED_CLASS = 'selected'

const Wrapper = styled.li`
  display: inline-block;
  white-space: nowrap;
  list-style: none;
  &::after {
    padding-right: 0.5em;
    padding-left: 0.5em;
    color: ${get('colors.gray.2')};
    font-size: ${get('fontSizes.1')};
    content: '/';
  }
  &:first-child {
    margin-left: 0;
  }
  &:last-child {
    &::after {
      content: none;
    }
  }
`

const BreadcrumbBase = styled.nav<SystemFlexProps & SystemCommonProps & SxProp>`
  display: flex;
  justify-content: space-between;
  ${COMMON};
  ${FLEX};
  ${sx};
`

export type BreadcrumbProps = ComponentProps<typeof BreadcrumbBase>

function Breadcrumb({className, children, theme, ...rest}: React.PropsWithChildren<BreadcrumbProps>) {
  const classes = classnames(className, 'Breadcrumb')
  const wrappedChildren = React.Children.map(children, child => <Wrapper theme={theme}>{child}</Wrapper>)
  return (
    <BreadcrumbBase className={classes} aria-label="breadcrumb" theme={theme} {...rest}>
      <Box as="ol" my={0} pl={0}>
        {wrappedChildren}
      </Box>
    </BreadcrumbBase>
  )
}

type StyledBreadcrumbItemProps = {
  to?: History.LocationDescriptor
  selected?: boolean
} & SystemCommonProps &
  SxProp

const BreadcrumbItem = styled.a.attrs<StyledBreadcrumbItemProps>(props => ({
  activeClassName: typeof props.to === 'string' ? 'selected' : '',
  className: classnames(props.selected && SELECTED_CLASS, props.className),
  'aria-current': props.selected ? 'page' : null
}))<StyledBreadcrumbItemProps>`
  color: ${get('colors.blue.5')};
  display: inline-block;
  font-size: ${get('fontSizes.1')};
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
  &.selected {
    color: ${get('colors.gray.7')};
    pointer-events: none;
  }
  ${COMMON}
  ${sx};
`

Breadcrumb.defaultProps = {
  theme
}

Breadcrumb.propTypes = {
  ...COMMON.propTypes,
  ...sx.propTypes
}

Breadcrumb.displayName = 'Breadcrumb'

BreadcrumbItem.defaultProps = {
  theme
}

BreadcrumbItem.propTypes = {
  href: PropTypes.string,
  selected: PropTypes.bool,
  ...sx.propTypes,
  ...COMMON.propTypes
}

BreadcrumbItem.displayName = 'Breadcrumb.Item'

export type BreadcrumbItemProps = ComponentProps<typeof BreadcrumbItem>
export default Object.assign(Breadcrumb, {Item: BreadcrumbItem})
