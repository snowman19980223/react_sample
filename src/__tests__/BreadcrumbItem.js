import React from 'react'
import Breadcrumb from '../Breadcrumbs'
import {render} from '../utils/testing'
import {render as HTMLRender, cleanup} from '@testing-library/react'
import {axe, toHaveNoViolations} from 'jest-axe'
import 'babel-polyfill'
expect.extend(toHaveNoViolations)

describe('Breadcrumb.Item', () => {
  it('renders an <a> by default', () => {
    expect(render(<Breadcrumb.Item />).type).toEqual('a')
  })

  it('has default theme', () => {
    expect(Breadcrumb.Item).toSetDefaultTheme()
  })

  it('should have no axe violations', async () => {
    const {container} = HTMLRender(<Breadcrumb.Item />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
    cleanup()
  })

  it('renders the given "as" prop', () => {
    const Type = ({theme, ...props}) => <b {...props} />
    expect(render(<Breadcrumb.Item as={Type} />)).toMatchSnapshot()
  })

  it('respects the "selected" prop', () => {
    expect(render(<Breadcrumb.Item selected />)).toMatchSnapshot()
  })

  it('adds activeClassName={SELECTED_CLASS} when it gets a "to" prop', () => {
    const Mock = jest.fn(() => <div />)
    render(<Breadcrumb.Item as={Mock} to="#" />)
    expect(Mock.mock.calls[0][0].to).toEqual('#')
    expect(Mock.mock.calls[0][0].activeClassName).toEqual('selected')
  })
})
