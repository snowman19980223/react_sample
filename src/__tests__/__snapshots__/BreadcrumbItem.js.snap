// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb.Item renders the given "as" prop 1`] = `
.c0 {
  color: #0366d6;
  display: inline-block;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0.selected {
  color: #444d56;
  pointer-events: none;
}

<b
  activeClassName=""
  aria-current={null}
  className="c0"
  theme={
    Object {
      "borders": Array [
        0,
        "1px solid",
      ],
      "breakpoints": Array [
        "544px",
        "768px",
        "1012px",
        "1280px",
      ],
      "buttons": Object {
        "danger": Object {
          "bg": Object {
            "active": "#be222e",
            "default": "#fafbfc",
            "disabled": "#F3F4F6",
            "hover": "#cb2431",
          },
          "border": Object {
            "active": "#9e1c23",
            "default": "#e1e4e8",
            "hover": "#b31d28",
          },
          "color": Object {
            "active": "#fff",
            "default": "#cb2431",
            "disabled": "rgba(203,36,49, .5)",
            "hover": "#fff",
          },
          "shadow": Object {
            "active": "inset 0px 2px 0px rgba(179, 29, 40, 0.4)",
            "default": "0px 1px 0px rgba(149, 157, 165, 0.1), inset 0px 2px 0px rgba(255, 255, 255, 0.25)",
            "focus": "0 0 0 3px rgba(203, 36, 49, 0.4)",
            "hover": "0px 1px 0px rgba(149, 157, 165, 0.1), inset 0px 2px 0px rgba(255, 255, 255, 0.03)",
          },
        },
        "default": Object {
          "bg": Object {
            "active": "#edeff2",
            "default": "#fafbfc",
            "disabled": "#fafbfc",
            "hover": "#F3F4F6",
          },
          "border": Object {
            "active": "#d1d5da",
            "default": "#e1e4e8",
            "disabled": "#eaecef",
          },
          "color": Object {
            "default": "#24292e",
            "disabled": "#959da5",
          },
          "shadow": Object {
            "active": "inset 0px 2px 0px rgba(149, 157, 165, 0.1)",
            "default": "0px 1px 0px rgba(149, 157, 165, 0.1), inset 0px 2px 0px rgba(255, 255, 255, 0.25)",
            "focus": "0 0 0 3px rgba(3, 102, 214, 0.3)",
            "hover": "0px 1px 0px rgba(209, 213, 218, 0.2), inset 0px 2px 0px rgba(255, 255, 255, 0.1)",
          },
        },
        "outline": Object {
          "bg": Object {
            "active": "#035fc7",
            "default": "#fafbfc",
            "disabled": "#F3F4F6",
            "hover": "#0366d6",
          },
          "border": Object {
            "active": "rgba(4, 66, 137, .5)",
            "default": "#e1e4e8",
            "hover": "#005cc5",
          },
          "color": Object {
            "active": "#fff",
            "default": "#005cc5",
            "disabled": "#959da5",
            "hover": "#fff",
          },
          "shadow": Object {
            "active": "inset 0px 1px 0px rgba(4, 66, 137, 0.2)",
            "default": "0px 1px 0px rgba(149, 157, 165, 0.1), inset 0px 2px 0px rgba(255, 255, 255, 0.25)",
            "focus": "0 0 0 3px rgba(3, 102, 214, 0.3)",
            "hover": "0px 1px 0px rgba(149, 157, 165, 0.1), inset 0px 2px 0px rgba(255, 255, 255, 0.03)",
          },
        },
        "primary": Object {
          "bg": Object {
            "active": "#128031",
            "default": "#159739",
            "disabled": "#94D3A2",
            "focus": "#138934",
            "hover": "#138934",
          },
          "border": Object {
            "active": "#176f2c",
            "default": "#22863a",
            "disabled": "rgba(34, 134, 58, 0.1)",
            "hover": "#176f2c",
          },
          "color": Object {
            "default": "#fff",
            "disabled": "rgba(255, 255, 255, 0.50)",
          },
          "shadow": Object {
            "active": "inset 0px 1px 0px rgba(20, 70, 32, 0.2)",
            "default": " 0px 1px 0px rgba(20, 70, 32, 0.1), inset 0px 2px 0px rgba(255, 255, 255, 0.03)",
            "focus": "0 0 0 3px #94D3A2",
          },
        },
      },
      "colors": Object {
        "bg": Object {
          "disabled": "#F3F4F6",
          "gray": "#f6f8fa",
          "grayLight": "#fafbfc",
        },
        "black": "#1b1f23",
        "blackfade15": "rgba(27, 31, 35, 0.15)",
        "blackfade20": "rgba(27, 31, 35, 0.20)",
        "blackfade30": "rgba(27,31,35,0.3)",
        "blackfade35": "rgba(27, 31, 35, 0.35)",
        "blackfade50": "rgba(27, 31, 35, 0.5)",
        "blue": Array [
          "#f1f8ff",
          "#dbedff",
          "#c8e1ff",
          "#79b8ff",
          "#2188ff",
          "#0366d6",
          "#005cc5",
          "#044289",
          "#032f62",
          "#05264c",
        ],
        "bodytext": "#24292e",
        "border": Object {
          "gray": "#e1e4e8",
          "grayDark": "#d1d5da",
          "grayLight": "#eaecef",
        },
        "counter": Object {
          "bg": "rgba(27, 31, 35, 0.08)",
        },
        "filterList": Object {
          "hoverBg": "#eaecef",
        },
        "gray": Array [
          "#fafbfc",
          "#f6f8fa",
          "#e1e4e8",
          "#d1d5da",
          "#959da5",
          "#6a737d",
          "#586069",
          "#444d56",
          "#2f363d",
          "#24292e",
        ],
        "green": Array [
          "#f0fff4",
          "#dcffe4",
          "#bef5cb",
          "#85e89d",
          "#34d058",
          "#28a745",
          "#22863a",
          "#176f2c",
          "#165c26",
          "#144620",
        ],
        "orange": Array [
          "#fff8f2",
          "#ffebda",
          "#ffd1ac",
          "#ffab70",
          "#fb8532",
          "#f66a0a",
          "#e36209",
          "#d15704",
          "#c24e00",
          "#a04100",
        ],
        "purple": Array [
          "#f5f0ff",
          "#e6dcfd",
          "#d1bcf9",
          "#b392f0",
          "#8a63d2",
          "#6f42c1",
          "#5a32a3",
          "#4c2889",
          "#3a1d6e",
          "#29134e",
        ],
        "red": Array [
          "#ffeef0",
          "#ffdce0",
          "#fdaeb7",
          "#f97583",
          "#ea4a5a",
          "#d73a49",
          "#cb2431",
          "#b31d28",
          "#9e1c23",
          "#86181d",
        ],
        "state": Object {
          "error": "#d73a49",
          "failure": "#d73a49",
          "pending": "#dbab09",
          "queued": "#dbab09",
          "success": "#28a745",
          "unknown": "#959da5",
        },
        "text": Object {
          "gray": "#586069",
          "grayDark": "#24292e",
          "grayLight": "#6a737d",
          "red": "#cb2431",
        },
        "white": "#fff",
        "whitefade15": "rgba(255, 255, 255, 0.15)",
        "whitefade50": "rgba(255, 255, 255, 0.50)",
        "yellow": Array [
          "#fffdef",
          "#fffbdd",
          "#fff5b1",
          "#ffea7f",
          "#ffdf5d",
          "#ffd33d",
          "#f9c513",
          "#dbab09",
          "#b08800",
          "#735c0f",
        ],
      },
      "fontSizes": Array [
        "12px",
        "14px",
        "16px",
        "20px",
        "24px",
        "32px",
        "40px",
        "48px",
      ],
      "fontWeights": Object {
        "bold": 600,
        "light": 300,
        "normal": 400,
        "semibold": 500,
      },
      "fonts": Object {
        "mono": "SFMono-Regular, Consolas, \\"Liberation Mono\\", Menlo, Courier, monospace",
        "normal": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"",
      },
      "lineHeights": Object {
        "condensed": 1.25,
        "condensedUltra": 1,
        "default": 1.5,
      },
      "maxWidths": Object {
        "large": "1012px",
        "medium": "768px",
        "small": "544px",
        "xlarge": "1280px",
      },
      "radii": Array [
        "0",
        "3px",
        "6px",
        "100px",
      ],
      "shadows": Object {
        "extra-large": "0 10px 50px rgba(27, 31, 35, 0.07)",
        "formControl": "inset 0px 2px 0px rgba(225, 228, 232, 0.2)",
        "formControlDisabled": "inset 0px 2px 0px rgba(220, 227, 237, 0.3)",
        "formControlFocus": "rgba(3, 102, 214, 0.3) 0px 0px 0px 0.2em",
        "large": "0 1px 15px rgba(27, 31, 35, 0.15)",
        "medium": "0 1px 5px rgba(27, 31, 35, 0.15)",
        "primaryActiveShadow": "inset 0px 1px 0px rgba(20, 70, 32, 0.2)",
        "primaryShadow": "0px 1px 0px rgba(20, 70, 32, 0.1), inset 0px 2px 0px rgba(255, 255, 255, 0.03)",
        "small": "0 1px 1px rgba(27, 31, 35, 0.1)",
      },
      "space": Array [
        "0",
        "4px",
        "8px",
        "16px",
        "24px",
        "32px",
        "40px",
        "48px",
        "64px",
        "80px",
        "96px",
        "112px",
        "128px",
      ],
    }
  }
/>
`;

exports[`Breadcrumb.Item respects the "selected" prop 1`] = `
.c0 {
  color: #0366d6;
  display: inline-block;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0.selected {
  color: #444d56;
  pointer-events: none;
}

<a
  aria-current="page"
  className="c0 selected"
  selected={true}
/>
`;
